src/zarr/core/metadata/v2.py

```diff
@@ -108,6 +109,29 @@ def shards(self) -> ChunkCoords | None:
         return None
 
     def to_buffer_dict(self, prototype: BufferPrototype) -> dict[str, Buffer]:
+        def _serialize_fill_value(fv: Any) -> JSON:
```

Comment:
This new funciton is inside an already really long function definition (with another nested function underneath). To make it more readable, could you extract this function by itself? It would need to gain a `array_dtype` argument to replace to `self.dtype`, but I think that's much better than having these nested functions.